ClosedXML/Excel/Cells/XLCell.cs

```diff
@@ -655,79 +655,210 @@ public IXLTable InsertTable<T>(IEnumerable<T> data, string tableName)
             return InsertTable(data, tableName, true);
         }
 
-        public IXLTable InsertTable<T>(IEnumerable<T> data, string tableName, bool createTable)
+        public IXLTable InsertTable<T>(IEnumerable<T> data, String tableName, Boolean createTable)
+        {
+            return InsertTable(data, tableName, createTable, addHeadings: true, transpose: false);
+        }
+
+        public IXLTable InsertTable<T>(IEnumerable<T> data, String tableName, Boolean createTable, Boolean addHeadings, Boolean transpose)
         {
             if (createTable && this.Worksheet.Tables.Any(t => t.Contains(this)))
                 throw new InvalidOperationException(String.Format("This cell '{0}' is already part of a table.", this.Address.ToString()));
 
-            if (data != null && !(data is String))
+            var range = InsertDataInternal(data, addHeadings, transpose);
+
+            if (createTable)
+                // Create a table and save it in the file
+                return tableName == null ? range.CreateTable() : range.CreateTable(tableName);
+            else
+                // Create a table, but keep it in memory. Saved file will contain only "raw" data and column headers
+                return tableName == null ? range.AsTable() : range.AsTable(tableName);
+        }
+
+        public IXLTable InsertTable(DataTable data)
+        {
+            return InsertTable(data, null, true);
+        }
+
+        public IXLTable InsertTable(DataTable data, Boolean createTable)
+        {
+            return InsertTable(data, null, createTable);
+        }
+
+        public IXLTable InsertTable(DataTable data, String tableName)
+        {
+            return InsertTable(data, tableName, true);
+        }
+
+        public IXLTable InsertTable(DataTable data, String tableName, Boolean createTable)
+        {
+            if (data == null || data.Columns.Count == 0)
+                return null;
+
+            if (createTable && this.Worksheet.Tables.Any(t => t.Contains(this)))
+                throw new InvalidOperationException(String.Format("This cell '{0}' is already part of a table.", this.Address.ToString()));
+
+            if (data.Rows.Cast<DataRow>().Any()) return InsertTable(data.Rows.Cast<DataRow>(), tableName, createTable);
```

Comment:
`data.Rows.Cast<DataRow>()` will throw an exception in case of invalid cast, if I am not mistaken. `data.Rows.OfType<DataRow>()` would fit better.

Reply:
`data` is of type `DataTable` and `DataTable.Rows` should contain only items of type `DataRow`. I'd prefer an exception if, for some reason, this wasn't the case, rather than "losing" any items.

Reply:
You're right, I missed this is a `DataTable`